File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        std::vector<int> list1;
        list1.push_back(1);
        std::vector<vector<int>> ans;
        ans.push_back(list1);
        while(numRows>1){
            std::vector<int> list2;
            list2.push_back(1);
            for(int i=0;i<list1.size()-1;i++)
                list2.push_back(list1[i]+list1[i+1]);
            list2.push_back(1);
            list1 = list2;
            ans.push_back(list1);
            numRows--;
        }
        return ans;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> list1 = new ArrayList<>();
        list1.add(1);
        ans.add(list1);
        while(numRows>1){
            List<Integer> list2 = new ArrayList<>();
            list2.add(1);
            for(int i=0;i<list1.size()-1;i++)
                list2.add(list1.get(i)+list1.get(i+1));
            list2.add(1);
            ans.add(list2);
            list1 = list2;
            numRows--;
        }
        return ans;
    }
}